src/utils/searchForm.util.js

```diff
@@ -79,6 +80,31 @@ const getFormData = (req, resourceName) => {
   return formData;
 };
 
+const getAdvSearchFormData = (req, resourceName) => {
+  const basicFormData = getFormData(req, resourceName);
+  const searchParams = getSearchParams(req);
+  const params = searchParams ? Object.fromEntries(searchParams.entries()) : {};
+  let advFormData = [];
+  const resourceFields = advSearchJson.entry.filter((entry) => {
+    return entry.resource.base.includes(resourceName) && entry.resource.type === 'string';
+  });
+
+  resourceFields.forEach((advParam) => {
+    const foundBasic = basicFormData.find((formData) => formData.name === advParam.resource.name);
+    if (!foundBasic) {
```

Comment:
Could we invert the if here to be if (foundBasic) return;, to reduce nesting

Reply:
Updated